src/AI/agents/agents/services/llm/intent_parser.py

```
@@ -155,10 +155,41 @@ def _validate_goal_safety_quick(goal: str) -> tuple[bool, Optional[str]]:
+async def suggest_goal_correction(
+    goal: str, rejection_reason: Optional[str] = None
+) -> List[str]:
+    """Suggest goals the user could ask for instead of the rejected one.
+
+    Never raises: a rejection must not turn into a generic error because the
+    suggestions could not be produced.
+    """
     try:
-        return suggest_goals_with_llm(goal)
+        candidates = suggest_goals_with_llm(goal, rejection_reason)
     except Exception as e:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)